app/src/main/kotlin/com/flxrs/dankchat/chat/ChatAdapter.kt

```diff
@@ -706,6 +711,14 @@ class ChatAdapter(
     private fun String.toRequest(context: Context): ImageRequest = ImageRequest.Builder(context)
         .data(this)
         .build()
+
+    /** set background color, and enable/disable ripple (whether enable or disable should match the "clickablity" of that message */
+    private fun TextView.setRippleBackground(backgroundColor: Int, enableRipple: Boolean = false) {
+        (background as RippleDrawable).apply {
```

Comment:
This apply is not needed. Can we be sure that the background is always a RippleDrawable? Otherwise I would suggest to use something like:
```kotlin
if (background !is RippleDrawable) {
    // old setBackground logic here and early return
}
...
```

Reply:
Yes, we can be sure the background is always RippleDrawable. It's set in XML layout file.

Why do we don't need the apply tho?

Reply:
I'll add comment explaining why it's always RippleDrawable 😀

Reply:
Sure, but that ripple is defined in another file at a completely different path. So if something changes this cast can easily be forgotten to adapt.
I just want to avoid hard casts wherever possible

Reply:
I see, I'll make it fallback to setting the background only, something like
```
background = ColorDrawable(bgColor)
```